include/nbl/builtin/hlsl/type_traits.hlsl

```diff
@@ -560,7 +559,7 @@ template<class T, unsigned I = 0>
 using extent = std::extent<T, I>;
 
 template<typename T>
-struct typeid_t : std::integral_constant<uint64_t,typeid(T).hash_code()> {};
+struct typeid_t : std::integral_constant<uint64_t,0> {};
```

Comment:
ok we need to:
- get rid of a C++ `typeid_t` cause `typeid()` is a runtime thing in C++ 
- and move the HLSL `typeid_t` to `impl` namespace
- implement a `nbl::hlsl::type_info` for HLSL and alias it to `std::type_info`

Reply:
Ok this is for another PR

Reply:
these two I need before merging:
- get rid of a C++ typeid_t cause typeid() is a runtime thing in C++
- and move the HLSL typeid_t to impl namespace